src/mrpro/utils/_change_data_shape.py

```diff
@@ -0,0 +1,49 @@
+"""Wrapper for FFT and IFFT."""
+
+# Copyright 2023 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+
+
+def change_data_shape(dat: torch.Tensor, dat_shape_new: tuple[int, ...]) -> torch.Tensor:
```

Comment:
couple of possible issues:
1. name sounds like a reshape operation.
2. consider data instead of data
3. consider dat_shape_new -> new_shape (or I might call you yoda)
4. what if data has a different number of dimensions than the new shape?

```suggestion
def pad_or_crop(data: torch.Tensor, new_shape: tuple[int, ...]|torch.Size) -> torch.Tensor:
    """Change shape of data by cropping or zero-padding.

    Parameters
    ----------
    data
        data
    new_shape
        target shape of data

    Returns
    -------
        data padded or cropped to shape
    """

    if len(new_shape) != data.ndim:
        raise ValueError("length of shape should match dimensions of data")
    npad=[]
    for old, new in zip(data.shape, new_shape):
        diff = new-old
        #  ensures padding and cropping lead to the matching asymmetry for odd shape differences
        before = math.trunc(diff/2) 
        after = diff - before
        npad.append(before)
        npad.append(after)
    # npad has to be reversed because pad expects it in reversed order
    if any(npad):
        data = F.pad(data, npad[::-1])
    return data
```

Reply:
I incorporated your second suggestion with the dimension option to be able to use it direclty in the PadOp